bpfassets/libbpf/src/kepler.bpf.c

```diff
@@ -109,19 +108,18 @@ static inline u64 get_on_cpu_time(u32 cur_pid, u32 prev_pid, u64 cur_ts)
 			bpf_map_delete_elem(&pid_time, &prev_pid_key);
 		}
 	}
-	new_pid_key.pid = cur_pid;
+
 	bpf_map_update_elem(&pid_time, &new_pid_key, &cur_ts, BPF_NOEXIST);
 	return cpu_time;
 }
 
-static inline u64 calc_delta(u64 *prev_val, u64 *val)
+static inline u64 calc_delta(u64 *prev_val, u64 val)
```

Comment:
note that the current implementation already assumes `u64 *val` is not nil.

Reply:
while you're here wouldn't it be better to make this `calc_delta(u64 prev_val, u64 val)`?